gum/backend-qnx/gumprocess-qnx.c

```diff
@@ -447,14 +448,80 @@ gum_module_enumerate_exports (const gchar * module_name,
         switch (dyn_entry->d_tag)
         {
           case DT_STRTAB:
-            dynsym_strtab = base_address + dyn_entry->d_un.d_ptr;
+            dynsym_strtab = (ehdr->e_type == ET_EXEC ? 0 : base_address)
+                + dyn_entry->d_un.d_ptr;
             break;
           case DT_SYMTAB:
             dynsym_section_offset = dyn_entry->d_un.d_ptr;
             break;
+          case DT_GNU_HASH:
+          {
+            guint * dyn_gnu_hash = (guint *) (
+                (ehdr->e_type == ET_EXEC ? 0 : base_address)
+                + dyn_entry->d_un.d_ptr);
+            guint nbuckets = dyn_gnu_hash[0];
+            guint symndx = dyn_gnu_hash[1];
+            guint bitmaskwords = dyn_gnu_hash[2];
+            guint buckets_vma = dyn_entry->d_un.d_ptr + 16 + bitmaskwords * 4;
+            guint * gnu_buckets = (guint *) (
+                (ehdr->e_type == ET_EXEC ? 0 : base_address) + buckets_vma);
+            guint maxchain = -1;
+            guint i;
+            guint * dynamic_info;
+            guint * gnu_chains;
+            guint hn;
```

Comment:
I think it would be cleaner if we separated the declarations from the assignments here, just smashing the declarations together in the same order that they're assigned below, with a blank line between the declarations and the assignments.